sys/kern/proc.c

```diff
@@ -339,64 +339,72 @@ static bool is_zombie(proc_t *p) {
   return p->p_state == PS_ZOMBIE;
 }
 
+static bool child_need_attention(proc_t *child, pid_t pid, pgrp_t *pg) {
+  /* pid > 0 => child with PID same as pid */
+  if (pid == child->p_pid)
+    return true;
+  /* pid == -1 => any child  */
+  if (pid == -1)
+    return true;
+  /* pid == 0 => child with PGID same as ours */
+  if (pid == 0 && child->p_pgrp == pg)
+    return true;
+  /* pid < -1 => child with PGID equal to -pid */
+  if (pid < -1 && child->p_pgrp->pg_id != -pid)
```

Comment:
There should probably be `==` instead of `!=` here.

Reply:
Fixed!